### PR TITLE
Sanitize styler icons and rework styler

### LINKED ISSUE #1010: Feature styler svg images are not being loaded
icons

### PR DESCRIPTION
**Pull request by raitisbe**
fixes #1010

### CODE REVIEW

**Comment by raitisbe** [comment]
[Release 2.4.0](https://trello.com/c/ZPfH6ews/106-release-240)

**Comment by raitisbe** [comment]
After sanitization I tried both with url-loader and file-loader. Still can't explain why it was working in other projects without sanitization before.

File: components/layermanager/partials/layermanagermd.html [hunk 1]
```diff
@@ -99,7 +99,7 @@
                     ng-if="layer.layer.getSource().styleAble">{{'LAYERMANAGER.layermanagerMd.layerStyle' | translate}}</label>
                 <div ng-if="layer.layer.getSource().hasPoly">
                     <div md-color-picker ng-model="layer.style.fillColor" label="Fill"
-                        ng-change="HsLayerEditorStylesService.HsLayerEditorStylesService.saveStyle(layer)"
```
Comment (jmacura) [inline components/layermanager/partials/layermanagermd.html:102]: Well spotted!

**Review by jmacura — CHANGES_REQUESTED** [review]
I really like the way you break up large functions into smaller private methods. The code is more readable then 👍 

I have tested the full example under this branch. The icons now do appear as they should.
But there is another issue now: the styler won't restyle features (neither polygons nor lines) until I click the "Icon" button in the Point section. The restyling of points happens immediately, as expected. But the "Circle" option doesn't seem to show anything at all...

**Comment by DailisLangovskis** [comment]
> I really like the way you break up large functions into smaller private methods. The code is more readable then 👍
> 
> I have tested the full example under this branch. The icons now do appear as they should.
> But there is another issue now: the styler won't restyle features (neither polygons nor lines) until I click the "Icon" button in the Point section. The restyling of points happens immediately, as expected. But the "Circle" option doesn't seem to show anything at all...

Fixed!

**Comment by jmacura** [comment]
> 
> > But there is another issue now: the styler won't restyle features (neither polygons nor lines) until I click the "Icon" button in the Point section. The restyling of points happens immediately, as expected. But the "Circle" option doesn't seem to show anything at all...
> 
> Fixed!

Thanks, a follow-up issue: all styling works, until I change "Level" from "Layer" to "Feature" and then back again. Then no styling works at all.

**Comment by jmacura** [comment]
The unit-tests are failing because of missing mock of HsUtilsService in styler.spec.ts. But you probably know that, right? 🙂

**Comment by jmacura** [comment]
> 
> Thanks, a follow-up issue: all styling works, until I change "Level" from "Layer" to "Feature" and then back again. Then no styling works at all.

I have tested it a bit more and it is a bit more clear now: The bug only happens with layers which are not clustered.

**Comment by raitisbe** [comment]
> layers which are not clustered.

Dhould be fixed now

**Comment by jmacura** [comment]
After the last commits: It works fine for "Circle" style now, but still not for "Icon" style.

My commits have added few more defaults.

**Review by raitisbe — COMMENTED** [review]
I think we can also read circle/icon type from the existing style same way as we do for color. Could be another issue together with reading of alpha, icon size and other properties.

**Review by raitisbe — COMMENTED** [review]
Well spotted

File: components/layermanager/layer-editor-vector-layer.service.ts [hunk 2]
```diff
@@ -115,39 +116,16 @@ export class HsLayerEditorVectorLayerService {
...
-              color: 'rgba(255,255,255,0.5)',
-            }),
-            stroke: new Stroke({
-              color: 'rgba(0, 153, 255, 1)',
-              width: 1.25,
-            }),
-          }),
-        });
-        originalStyle = defaultStyle;
-        layer.set('hsOriginalStyle', defaultStyle);
```
Comment (raitisbe) [inline components/layermanager/layer-editor-vector-layer.service.ts]: Can't set hsOriginalStyle as many times as there are features

File: components/styles/styler.component.ts [hunk 3]
```diff
@@ -25,21 +25,8 @@ type StyleJson = {
 })
 export class HsStylerComponent {
   icons: any;
-  imagetypes: Array<any> = [
```
Comment (raitisbe) [inline components/styles/styler.component.ts:25]: Originally this approach was used to support translations of "Icon"and "Circle". hrname could change, but underlying value would remain,.

Comment (jmacura) [reply components/styles/styler.component.ts:25]: Ah, I did not realised that.. Will revert my "improvement" then.

File: components/styles/styler.service.ts [hunk 4]
```diff
@@ -62,4 +63,20 @@ export class HsStylerService {
       }),
     ];
   }
+
+  /**
+   * @function save
+   * @memberof hs.styler.controller
+   * @description Get current style variables value and style current layer accordingly
```
Comment (FilipLeitner) [inline components/styles/styler.service.ts:70]: Incorrect description. Would change tjhat ust dont want to interfeere with unexpected commits

Comment (jmacura) [reply components/styles/styler.service.ts:70]: I have some commits on the go. I gonna squeeze this change between them 😉 Thanks for the tip.

File: components/styles/styler.component.ts (hunk 3 above)
Comment (jmacura) [reply components/styles/styler.component.ts]: Reverted.

File: components/styles/styler.service.ts (hunk 4 above)
Comment (jmacura) [reply components/styles/styler.service.ts]: Corrected in cbdaab6

**Review by jmacura — APPROVED** [review]
Seems good enough for me now.

Just a last question: is there a reason, why lines and polygons are styled with the same line-color and line-width, while points have their own style?

**Comment by raitisbe** [comment]
If I rememer correctly its because icon in openlayers expects a seperate config of colors and lines.

**Comment by DailisLangovskis** [comment]
> If I rememer correctly its because icon in openlayers expects a seperate config of colors and lines.

This is true, icon and circle uses image style which has its own stroke and fill colors

**Pull request merged**